tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/operations/selectors/LaunchBarTargetSelector.java

```diff
@@ -98,27 +106,80 @@ public LaunchBarTargetSelector select(String text)
 
 	public LaunchBarTargetSelector selectTarget(String text)
 	{
-		click();
-		SWTBotShell swtBotShell = bot().shellWithId(LaunchBarWidgetIds.POPUP);
-		ScrolledComposite scrolledComposite = swtBotShell.bot().widget(widgetOfType(ScrolledComposite.class));
-		int numberOfItemsInScrolledComp = syncExec(
-				() -> ((Composite) scrolledComposite.getChildren()[0]).getChildren().length);
-		Label itemToSelect;
-
-		// Set the text in the not visible text field
-		// when the target list is too big, swtbot cannot select a target label, so we filter the list
-		if (numberOfItemsInScrolledComp > NUM_FOR_FILTER_POPUP)
-		{
-			swtBotShell.bot().text().setText(text);
-			itemToSelect = swtBotShell.bot().label(0).widget;
-		}
-		else
-		{
-			itemToSelect = swtBotShell.bot().widget(withText(text));
-		}
-
-		Point itemToSelectLocation = syncExec((Result<Point>) itemToSelect::getLocation);
-		clickOnInternalWidget(itemToSelectLocation.x, itemToSelectLocation.y, itemToSelect);
-		return this;
+	    click();
+	    SWTBotShell swtBotShell = bot().shellWithId(LaunchBarWidgetIds.POPUP);
+	    ScrolledComposite scrolledComposite = swtBotShell.bot().widget(widgetOfType(ScrolledComposite.class));
+	    int numberOfItemsInScrolledComp = syncExec(
+	            () -> ((Composite) scrolledComposite.getChildren()[0]).getChildren().length);
+	    Label itemToSelect;
+
+	    if (numberOfItemsInScrolledComp > NUM_FOR_FILTER_POPUP)
+	    {
+	        swtBotShell.bot().text().setText(text);
+	        itemToSelect = swtBotShell.bot().widget(allOf(widgetOfType(Label.class), withText(text)));
+	    }
+	    else
+	    {
+	        itemToSelect = swtBotShell.bot().widget(allOf(widgetOfType(Label.class), withText(text)));
+	    }
+
+	    Point itemToSelectLocation = syncExec((Result<Point>) itemToSelect::getLocation);
+	    clickOnInternalWidget(itemToSelectLocation.x, itemToSelectLocation.y, itemToSelect);
+	    return this;
+	}
+
+	public void scrollToBottom(ScrolledComposite scrolledComposite)
+	{
+	    syncExec(() -> {
+	        scrolledComposite.setOrigin(0, scrolledComposite.getClientArea().height);
+	    });
+	}
+
+	public boolean isTargetPresent(String text)
+	{
+	    click();
+
+	    try
+	    {
+	        SWTBotShell swtBotShell = bot().shellWithId(LaunchBarWidgetIds.POPUP);
+	        ScrolledComposite scrolledComposite = swtBotShell.bot().widget(widgetOfType(ScrolledComposite.class));
+
+	        int numberOfItemsInScrolledComp = syncExec(() ->
+	            ((Composite) scrolledComposite.getChildren()[0]).getChildren().length
+	        );
+
+	        // Scroll to the bottom if there are many items
+	        if (numberOfItemsInScrolledComp > NUM_FOR_FILTER_POPUP)
+	        {
+	            scrollToBottom(swtBotShell.bot().widget(widgetOfType(ScrolledComposite.class)));
+	            swtBotShell.bot().text().setText(text);
+
+	            List<? extends Widget> labels = swtBotShell.bot().widgets(widgetOfType(Label.class));
```

Comment:
if you use ? extends Label you wont have to use the type casting later in the loop